app/src/main/java/com/vestrel00/daggerbutterknifemvp/ui/common/BaseActivity.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 Vandolf Estrellado
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.vestrel00.daggerbutterknifemvp.ui.common;
+
+import android.app.Activity;
+import android.app.Fragment;
+import android.app.FragmentManager;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import dagger.android.AndroidInjection;
+import dagger.android.AndroidInjector;
+import dagger.android.DispatchingAndroidInjector;
+import dagger.android.HasFragmentInjector;
+
+/**
+ * Abstract Activity for all Activities to extend.
+ * <p>
+ * <b>DEPENDENCY INJECTION</b>
+ * We could extend {@link dagger.android.DaggerActivity} so we can get the boilerplate
+ * dagger code for free. However, we want to avoid inheritance (if possible and it is in this case)
+ * so that we have to option to inherit from something else later on if needed.
+ */
+public abstract class BaseActivity extends Activity implements HasFragmentInjector {
+
+    @Inject
+    @Named(BaseActivityModule.ACTIVITY_FRAGMENT_MANAGER)
+    protected FragmentManager fragmentManager;
+
+    @Inject
+    DispatchingAndroidInjector<Fragment> fragmentInjector;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        AndroidInjection.inject(this);
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public final AndroidInjector<Fragment> fragmentInjector() {
+        return fragmentInjector;
+    }
+
+    protected final void addFragment(int containerViewId, Fragment fragment) {
```

Comment:
Nice catch! Thanks.

Fixed: https://github.com/vestrel00/android-dagger-butterknife-mvp/commit/88f9f5b4ee0afb6ea73cb7fd81bcd2d5934ffe23

I also added a quick tip about the support annotations in the article.